src/main/java/com/redhat/labs/lodestar/service/GitService.java

```diff
@@ -0,0 +1,337 @@
+
+	/**
+	 * Creates, updates, or deletes the {@link Artifact}s in the artifacts.json for
+	 * the given engagement ID. If any {@link Optional} parameter is not supplied
+	 * the configured default will be used.
+	 * 
+	 * @param engagementUuid
+	 * @param artifacts
+	 * @param authorEmail
+	 * @param authorName
+	 * @param commitMessage
+	 */
+	void createOrUpdateArtifactsFile(String engagementUuid, List<Artifact> artifacts, Optional<String> authorEmail,
+			Optional<String> authorName, Optional<String> commitMessage) {
+
+		// find project by engagement
+		Project project = findProjectByEngagementUuid(engagementUuid)
+				.orElseThrow(() -> new WebApplicationException("no project found with engagemnt id" + engagementUuid));
+
+		// does artifact file exist
+		boolean isUpate = artifactsFileExists(project);
+
+		// create json content
+		String content = jsonb.toJson(artifacts);
+
+		// find default branch
+		String branch = findDefaultBranch(project.getId());
+
+		// create
+		File artifactsFile = createArtifactsFile(content, branch, authorName, authorEmail, commitMessage);
+
+		// create or udpate in git
+		if (isUpate) {
+			gitlabRestClient.updateFile(project.getId(), artifactsFile.getFilePath(), artifactsFile);
+		} else {
+			gitlabRestClient.createFile(project.getId(), artifactsFile.getFilePath(), artifactsFile);
+		}
+
+	}
+
+	/**
+	 * Returns true if artifacts.json is found in the project's tree. Otherwise,
+	 * false.
+	 * 
+	 * @param project
+	 * @return
+	 */
+	boolean artifactsFileExists(Project project) {
+
+		// get project tree
+		ProjectTree projectTree = createProjectTree(project);
+
+		// update if file already exists
+		return projectTree.hasArtifactsJsonFile();
+
+	}
+
+	/**
+	 * Creates and returns a {@link File} with the given parameters and encodes the
+	 * content.
+	 * 
+	 * @param content
+	 * @param branch
+	 * @param authorName
+	 * @param authorEmail
+	 * @param commitMessage
+	 * @return
+	 */
+	File createArtifactsFile(String content, String branch, Optional<String> authorName, Optional<String> authorEmail,
+			Optional<String> commitMessage) {
+
+		// create file
+		File artifactFile = File.builder().filePath(ARTIFACT_FILE).content(content)
+				.authorEmail(authorEmail.orElse(defaultAuthorEmail)).authorName(authorName.orElse(defaultAuthorName))
+				.branch(branch).commitMessage(commitMessage.orElse(defaultCommitMessage)).build();
+
+		// encode before sending
+		artifactFile.encodeFileAttributes();
+
+		return artifactFile;
+
+	}
+
+	/**
+	 * Returns an {@link Optional} containing the {@link Project} found for the
+	 * given engagement ID. Otherwise, an empty {@link Optional} is returned.
+	 * 
+	 * @param uuid
+	 * @return
+	 */
+	Optional<Project> findProjectByEngagementUuid(String uuid) {
+
+		List<Project> projects = gitlabRestClient.findProjectByEngagementId(groupParentId, "projects", uuid);
+		if (1 != projects.size()) {
+			return Optional.empty();
+		}
+
+		return Optional.of(projects.get(0));
+
+	}
```

Comment:
May have already mentioned - I was expecting this to come from `git-api` which is standing in for the engagement api. When we move the engagements out of the backend it'll be simpler to switch to `lodestar-engagements` with the same functionality

Reply:
now calling git-api.